tests/Unit/Infrastructure/DonationConfirmationMailerTest.php

```diff
@@ -4,52 +4,61 @@
 
 namespace WMDE\Fundraising\Frontend\Tests\Unit\Infrastructure;
 
+use WMDE\Fundraising\Frontend\Domain\Model\Donation;
+use WMDE\Fundraising\Frontend\Domain\Model\EmailAddress;
 use WMDE\Fundraising\Frontend\Domain\Model\PaymentType;
 use WMDE\Fundraising\Frontend\Infrastructure\DonationConfirmationMailer;
-use WMDE\Fundraising\Frontend\Infrastructure\TemplateBasedMailer;
 use WMDE\Fundraising\Frontend\Tests\Data\ValidDonation;
+use WMDE\Fundraising\Frontend\Tests\Fixtures\TemplateBasedMailerSpy;
 
 class DonationConfirmationMailerTest extends \PHPUnit_Framework_TestCase {
 
+	const DONATION_ID = 42;
+
 	public function testMailerExtractsEmailFromDonation() {
+		$mailer = new TemplateBasedMailerSpy( $this );
+
+		$donationMailer = new DonationConfirmationMailer( $mailer );
+		$donationMailer->sendMailFromDonation( $this->newDonation() );
+
+		$mailer->assertCalledOnce();
+		$this->assertEquals(
+			new EmailAddress( ValidDonation::DONOR_EMAIL_ADDRESS ),
+			$mailer->getSendMailCalls()[0][0]
+		);
+	}
+
+	private function newDonation(): Donation {
 		$donation = ValidDonation::newBankTransferDonation();
-		$templateMailer = $this->getMockBuilder( TemplateBasedMailer::class )->disableOriginalConstructor()->getMock();
-		$templateMailer->expects( $this->once() )
-			->method( 'sendMail' )
-			->with(
-				$this->equalTo( ValidDonation::DONOR_EMAIL_ADDRESS ),
```

Comment:
Fun fact... this passes even though the argument is actually an `EmailAddress` object, as this check is not strict and casts the thing to string